### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by tingamapuro04**
I have worked in the validations branch where I
- added necessary validations
- added the necessary tests

### CODE REVIEW

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Alphonce', photo: 'https:/photo.com', bio: 'Teacher', posts_counter: 5) }
+
```
Comment (SajjadAhmad14) [inline spec/user_spec.rb:5]: Please add a test for `#recent_posts` in User model specs as any custom methods in the models should have corresponding unit tests.

File: spec/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  user = User.create(name: 'Alphonce', photo: 'https:/photo.com', bio: 'Teacher', posts_counter: 5)
+  subject { Post.new(author: user, title: 'Chinua', commentscounter: 10, likescounter: 7) }
+
```
Comment (SajjadAhmad14) [inline spec/post_spec.rb:6]: Please add tests for `#update_user_posts_count` and `#recent_comments` in Post model specs as any custom methods in the models should have corresponding unit tests.

File: spec/comments_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  user = User.create(name: 'Alphonce', photo: 'https:/photo.com', bio: 'Teacher', posts_counter: 5)
+  post1 = Post.create(author: user, title: 'Chinua', commentscounter: 10, likescounter: 7)
+
+  subject { Comment.new(post: post1, author: user, text: 'I am writing my first comment!') }
+  before { subject.save }
+
```
Comment (SajjadAhmad14) [inline spec/comments_spec.rb:9]: Please add a test for `#update_comments_counter` in Comment model specs as any custom methods in the models should have corresponding unit tests.

File: spec/like_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  describe 'Tests for Like model' do
+    usr = User.create(name: 'Ram', photo: 'https://unsplash.com/photos/6PP8DrhY0aA', bio: 'Hindu god.')
+    post = Post.create(author: usr, title: 'Intro', text: 'Welcome to this blog!')
+
+    subject { Like.new }
+    before { subject.save }
+
```
Comment (SajjadAhmad14) [inline spec/like_spec.rb:10]: Please add a test for `#update_likes_counter` in Like model specs as any custom methods in the models should have corresponding unit tests.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Alphonce, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by tingamapuro04** [comment]
I have added all the tests recommended above

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @tingamapuro04 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**